checker/check_result.go

```diff
@@ -63,6 +63,32 @@ const (
 	FileTypeURL
 )
 
+// OutcomeType represents the outcome of a result.
+type OutcomeType int
+
+const (
+	// OutcomeTypeUnknown is for unknown type. Used for debugging and catching unset types.
+	OutcomeTypeUnknown OutcomeType = iota
```

Comment:
Nit: type is already `OutcomeType`, so rename the enums to `Unknown`, `Debug` etc. to avoid redundancy.

Also, note that when this gets printed out it'll only print out an integer. Consider using a string type to make this clearer.